src/components/internal/DatePicker.stories.tsx

```diff
@@ -10,5 +10,5 @@ export default {
 
 export function Default() {
   const [date, setDate] = useState(jan1);
-  return <DatePicker value={date} onSelect={setDate} />;
+  return <DatePicker value={date} onSelect={setDate} indicators={[jan1, jan2, jan29]} />;
```

Comment:
Wdyt of having a callback sort of UI like `isIndicated(day) => boolean` so we don't have to materialize a full list of dates? Disclaimer I'm not really sure what this is used for, like will this list be small / medium / large? If we know it's always small, this is a great.

Reply:
This was an ask for the Job Logs page. When looking at the calendar, they want this indicator to show which days already have a Job Log attached. So this list could get pretty large as we expect to have a Job Log for _every_ working day.

Great question regarding how this should be used from Blueprint side of things as we probably wouldn't want to front load every date that has a job log. Since the Date picker displays month by month, maybe thats how we can do the API?

```
const [indicatedDates, setDates] = useState(data.logs.map((log) => log.date));
<DatePicker 
  onMonthChange={async (d: Date) => {
    // add some sort of caching to not query again if we already have this month's data.
    const result = await getJobLogsByMonth(); 
    setDates((prev) => prev.concat(result.data.logs.map((log) => log.date)));
  }};
  indicators={indicatedDates}
/>
```
Something like that? The list could still get really large, but hopefully not initially.

Reply:
@zgavin just checking, does this "each time the calendar width's month changes, we'll load the X days within that month have already have a job log"?

Or were you just going to load them all up-front anyway b/c :shrug: it'll be a few hundred and not a big deal...

Reply:
right now there is no pagination and we are just uploading them all up front.

Reply:
Cool, we could ship `indicators?: Date[]` today then, and later follow up with an `indicators?: Date[] | (month: Month) => MaybePromise<Date[]>` overload when/if we need it.